hack/ginkgo-e2e.sh

```diff
@@ -167,15 +167,26 @@ PATH=$(dirname "${e2e_test}"):"${PATH}"
 export PATH
 
 # Choose the program to execute.
-program=("${ginkgo}")
+program=("${ginkgo}"
+         --slow-spec-threshold="${GINKGO_SLOW_SPEC_THRESHOLD:-300s}"
+         --poll-progress-after="${GINKGO_POLL_PROGRESS_AFTER:-300s}"
+         --poll-progress-interval="${GINKGO_POLL_PROGRESS_INTERVAL:-20s}"
+         --source-root="${KUBE_ROOT}"
+)
 if [[ "${E2E_TEST_DEBUG_TOOL}" == "delve" ]]; then
   program=("dlv" "exec")
 elif [[ "${E2E_TEST_DEBUG_TOOL}" == "gdb" ]]; then
   program=("gdb")
 fi
 
+# The following invocation is fairly complex. Let's dump it to simplify
+# determining what the final options are.
+set -x
+
 # NOTE: Ginkgo's default timeout has been reduced from 24h to 1h in V2, set it manually here as "24h"
-# for backward compatibility purpose.
+# for backward compatibility purpose. These (few) Ginkgo parameters are passed here
+# to ensure that they are also active when not using the Ginkgo CLI, but in general arguments
+# are more likely to work when specified above.
 "${program[@]}" "${ginkgo_args[@]:+${ginkgo_args[@]}}" "${e2e_test}" -- \
   "${auth_config[@]:+${auth_config[@]}}" \
   --ginkgo.flake-attempts="${FLAKE_ATTEMPTS}" \
```

Comment:
anf this one and ginkgo.timeout?

Reply:
If we move `timeout`, then `dlv test ./test/e2e` will time out after one hour, the default timeout. Perhaps that is okay?
What we are doing now prevents using using a reasonable timeout in `ginkgo --timeout` because it would just get overwritten by the parameter we pass here - it has to be, or this parameter here wouldn't have had an effect.

Considering all this, I am leaning towards moving *all* Ginkgo parameters to the CLI invocation. It's simply cleaner.

Reply:
Or be even more intelligent: add them to the CLI when using it, otherwise to the suite parameters.

Reply:
I think that's the cleanest approach. I've pushed version with that and tried it locally with the Ginkgo CLI and delve.